hyperspy/model.py

```diff
@@ -210,50 +209,64 @@ class BaseModel(list):
 
     Attributes
     ----------
-
-    signal : BaseSignal instance
-        It contains the data to fit.
-    chisq : :py:class:`~.signal.BaseSignal` of float
-        Chi-squared of the signal (or np.nan if not yet fit)
-    dof : :py:class:`~.signal.BaseSignal` of int
-        Degrees of freedom of the signal (0 if not yet fit)
-    components : :py:class:`~.model.ModelComponents` instance
-        The components of the model are attributes of this class. This provides
-        a convenient way to access the model components when working in IPython
-        as it enables tab completion.
+    signal : :class:`~.api.signals.BaseSignal`
```

Comment:
We've discussed this already. But these lists of attributes and methods could be automatically generated from their docstrings using a custom Sphinx/Jinja2 template (example from kikuchipy's docs: [class template](https://github.com/pyxem/kikuchipy/blob/develop/doc/_templates/custom-class-template.rst) -> [result](https://kikuchipy.org/en/stable/reference/generated/kikuchipy.detectors.EBSDDetector.html)).

Reply:
I see that they suggest to list them if there are only a few out of many methods that are relevant:

> In some cases, however, a class may have a great many methods, of which only a few are relevant (e.g., subclasses of ndarray). Then, it becomes useful to have an additional Methods section

I counted 30 methods without a leading underscore in `BaseModel`, of which you have listed 20. So I guess this holds here :) 

A drawback of this section is duplication of information (names and descriptions), so I don't use it myself.